### PR TITLE
Allow admin to leave chat

### PR DESCRIPTION
**Pull request by cammellos**
### Description

partially covers: #4868 (only for group chats)

1) Allow admin to leave a chat (should not receive anymore messages)
2) Allow admin to make other members admin
3) Display system messages when a user is made admin

### Testing

To make a member an admin:
1) Create a group chat
2) Go to Group-Info -> Click on Dots -> Make admin

If the user is already an admin, you will only see "View profile"

When leaving a chat, if you are an admin, you will be removed from the chat and will not receive any more messages (previous version will still show the user as admin).

status: ready

### CODE REVIEW

**Comment by lukaszfryc** [comment]
I think this needs to go into release/0.9.32 as it contains a fix for https://github.com/status-im/status-react/issues/4868#issuecomment-441301900. 

cc https://github.com/status-im/status-react/issues/6891

**Comment by cammellos** [comment]
@lukaszfryc sure, if there's any issue etc I am happy to split off the PR for the fix (it's a one line change) and just get that in the release

**Comment by lukaszfryc** [comment]
The spinner issue is fixed. But, when I make another user an admin and then go to group info screen once again and click on that user, the app crashes. Crash log:

[StatusIm  29-11-2018, 12-19.txt](https://github.com/status-im/status-react/files/2628870/StatusIm.29-11-2018.12-19.txt)

@cammellos if it's not easy to fix, please consider splitting the PR.

**Comment by cammellos** [comment]
@lukaszfryc difficult to say what it is, can't replicate locally, will split the fix for now and have a look at it later

**Comment by cammellos** [comment]
https://github.com/status-im/status-react/pull/6935

**Comment by lukaszfryc** [comment]
I'm removing release from this PR label then.

File: src/status_im/ui/screens/profile/group_chat/views.cljs [hunk 1]
```diff
@@ -52,17 +52,24 @@
      :action              #(re-frame/dispatch [:group-chats.ui/remove-chat-pressed chat-id])
      :accessibility-label :delete-chat-button}]))
 
-(defn member-actions [chat-id member]
+(defn member-actions [chat-id member us-admin?]
```
Comment (yenda) [inline src/status_im/ui/screens/profile/group_chat/views.cljs:55]: is-admin?

Comment (cammellos) [reply src/status_im/ui/screens/profile/group_chat/views.cljs:55]: I though about that, but I wanted to convey the fact that it's us, not the member, and `is-admin?` might be a bit confusing, any other suggestion?

Comment (yenda) [reply src/status_im/ui/screens/profile/group_chat/views.cljs:55]: but who is `us`?

Comment (cammellos) [reply src/status_im/ui/screens/profile/group_chat/views.cljs:55]: `us` is `us` :) I mean it's the current user (`current-public-key`) :)

**Comment by rasom** [comment]
@cammellos What happens if nobody has been granted as admin?

**Comment by cammellos** [comment]
@rasom nobody will be able to add/remove members in that group chat, before this we did not allow the last admin to leave, but it was decided to change the behaviour until a better solution is found

**Comment by lukaszfryc** [comment]
@cammellos is it ready to be re-tested?

**Comment by cammellos** [comment]
@lukaszfryc maybe quickly try it on IOS, I could not replicate on android builds, but rebased against develop, didn't have time to look into it, also if you could replicate, do you have any other logs?

**Comment by lukaszfryc** [comment]
@cammellos still happens on iOS. I'm testing on iPhone 8, iOS 12.1. If I give admin to some user and then reopen group info screen and click on the "...", the app crashes.



More logs:

```
Exception '-[NSNull length]: unrecognized selector sent to instance 0x242ec19b0' was thrown while invoking showActionSheetWithOptions on target ActionSheetManager with params (
        {
        cancelButtonIndex = 3;
        options =         (
            "View profile",
            "<null>",
            "<null>",
            Cancel
        );
        title = "Cool Biodegradable Portuguesemanofwar";
    },
    7675
)
callstack: (
	0   CoreFoundation                      0x000000021275feb8 <redacted> + 252
	1   libobjc.A.dylib                     0x0000000211931a40 objc_exception_throw + 56
	2   CoreFoundation                      0x0000000212678d04 <redacted> + 0
	3   CoreFoundation                      0x00000002127657b8 <redacted> + 1412
	4   CoreFoundation                      0x000000021276745c _CF_forwarding_prep_0 + 92
	5   UIKitCore                           0x000000023f080838 <redacted> + 180
	6   UIKitCore                           0x000000023f080ad4 <redacted> + 148
	7   UIKitCore        <…>
```

**Comment by cammellos** [comment]
@lukaszfryc those are exactly the logs I needed! Will fix soon , thanks!

**Comment by cammellos** [comment]
@lukaszfryc could not make sure it works, but should be fixed in the new build (still building)

**Comment by lukaszfryc** [comment]
restarting the build as it failed (iOS build did not finish)

**Comment by lukaszfryc** [comment]
@cammellos I tested iOS, Android and macOS. All works fine including deleting the chat.

**Comment by lukaszfryc** [comment]
@cammellos please cherry pick to release/0.9.32

**Review by yenda — APPROVED** [review]

**Comment by cammellos** [comment]
@rasom could you review this please? It has been tested-ok and cherry-picked for release

**Review by goranjovic — APPROVED** [review]

**Pull request merged**